packages/contracts-bedrock/scripts/forge-test-names.ts

```diff
@@ -0,0 +1,81 @@
+import fs from 'fs'
+import path from 'path'
+
+const testPath = './contracts/test'
+const testFiles = fs.readdirSync(testPath)
+
+// Given a test function name, ensures it matches the expected format
+const handleFunctionName = (name: string) => {
+  if (!name.startsWith('test')) {
+    return
+  }
+  const parts = name.split('_')
+  parts.forEach((part) => {
+    // Good enough approximation for camelCase
+    if (part[0] !== part[0].toLowerCase()) {
+      throw new Error(
+        `Invalid test name: ${name}.\n Test name parts should be in camelCase`
+      )
+    }
+  })
+  if (parts.length < 3 || parts.length > 4) {
+    throw new Error(
+      `Invalid test name: ${name}.\n Test names should have either 3 or 4 parts, each separated by underscores`
+    )
+  }
+  if (!['test', 'testFuzz', 'testDiff'].includes(parts[0])) {
+    throw new Error(
+      `Invalid test name: ${name}.\n Names should begin with either "test" or "testFuzz"`
+    )
+  }
+  if (
+    !['succeeds', 'reverts', 'fails', 'benchmark', 'works'].includes(
+      parts[parts.length - 1]
+    ) &&
+    parts[parts.length - 2] !== 'benchmark'
+  ) {
+    throw new Error(
+      `Invalid test name: ${name}.\n Test names should end with either "succeeds", "reverts", "fails", "differential" or "benchmark[_num]"`
+    )
+  }
+  if (
+    ['reverts', 'fails'].includes(parts[parts.length - 1]) &&
+    parts.length < 4
+  ) {
+    throw new Error(
+      `Invalid test name: ${name}.\n Failure tests should have 4 parts. The third part should indicate the reason for failure.`
+    )
+  }
+}
+
+// Todo: define this function for validating contract names
+// Given a test contract name, ensures it matches the expected format
+const handleContractName = (name: string) => {
+  name
+}
+
+for (const testFile of testFiles) {
```

Comment:
I believe all of the test functions are marked `external` or `public` which means we could also use the ABI instead of scanning files line by line. I'm not particularly opinionated but using ABI might be slightly more robust.

Reply:
OK, I did that [here](https://github.com/ethereum-optimism/optimism/blob/9ae69986d242dffa246a8c3ccc6dc5a7695388c6/packages/contracts-bedrock/scripts/forge-test-names.ts#L90-L107).

I'm not sure it's the best way, but it gets the job done. Also, fwiw, I do hope that this is a temporary solution which we will eventually replace with a proper tool for enforcing our own conventions.